src/measure_theory/integral/set_to_l1.lean

```diff
@@ -662,6 +729,121 @@ begin
+  `set_to_fun`.
+  We could weaken the condition `bound_integrable` to require `has_finite_integral bound μ` instead
+  (i.e. not requiring that `bound` is measurable), but in all applications proving integrability
+  is easier. -/
+theorem tendsto_set_to_fun_of_dominated_convergence (hT : dominated_fin_meas_additive μ T C)
+  {fs : ℕ → α → E} {f : α → E} (bound : α → ℝ) (fs_measurable : ∀ n, ae_measurable (fs n) μ)
+  (bound_integrable : integrable bound μ) (h_bound : ∀ n, ∀ᵐ a ∂μ, ∥fs n a∥ ≤ bound a)
+  (h_lim : ∀ᵐ a ∂μ, tendsto (λ n, fs n a) at_top (𝓝 (f a))) :
+  tendsto (λ n, set_to_fun hT (fs n)) at_top (𝓝 $ set_to_fun hT f) :=
+begin
+  /- `f` is a.e.-measurable, since it is the a.e.-pointwise limit of a.e.-measurable functions. -/
+  have f_measurable : ae_measurable f μ := ae_measurable_of_tendsto_metric_ae fs_measurable h_lim,
+  /- all functions we consider are integrable -/
+  have fs_int : ∀ n, integrable (fs n) μ :=
+    λ n, bound_integrable.mono' (fs_measurable n) (h_bound _),
+  have f_int : integrable f μ :=
+  ⟨f_measurable, has_finite_integral_of_dominated_convergence
+    bound_integrable.has_finite_integral h_bound h_lim⟩,
+  /- it suffices to prove the result for the corresponding L1 functions -/
+  suffices : tendsto (λ n, L1.set_to_L1 hT ((fs_int n).to_L1 (fs n))) at_top
+    (𝓝 (L1.set_to_L1 hT (f_int.to_L1 f))),
+  { convert this,
+    { ext1 n, exact set_to_fun_eq hT (fs_int n), },
+    { exact set_to_fun_eq hT f_int, }, },
+  /- the convergence of set_to_L1 follows from the convergence of the L1 functions -/
+  refine L1.tendsto_set_to_L1 hT _ _ _,
+  /- up to some rewriting, what we need to prove is `h_lim` -/
+  rw tendsto_iff_norm_tendsto_zero,
+  have lintegral_norm_tendsto_zero :
+    tendsto (λn, ennreal.to_real $ ∫⁻ a, (ennreal.of_real ∥fs n a - f a∥) ∂μ) at_top (𝓝 0) :=
+  (tendsto_to_real zero_ne_top).comp
+    (tendsto_lintegral_norm_of_dominated_convergence
+      fs_measurable bound_integrable.has_finite_integral h_bound h_lim),
+  convert lintegral_norm_tendsto_zero,
+  ext1 n,
+  rw L1.norm_def,
+  congr' 1,
+  refine lintegral_congr_ae _,
+  rw ← integrable.to_L1_sub,
+  refine ((fs_int n).sub f_int).coe_fn_to_L1.mono (λ x hx, _),
+  dsimp only,
+  rw [hx, of_real_norm_eq_coe_nnnorm, pi.sub_apply],
+end
+
+/-- Lebesgue dominated convergence theorem for filters with a countable basis -/
+lemma tendsto_set_to_fun_filter_of_dominated_convergence (hT : dominated_fin_meas_additive μ T C)
+  {ι} {l : _root_.filter ι} [l.is_countably_generated]
+  {fs : ι → α → E} {f : α → E} (bound : α → ℝ)
+  (hfs_meas : ∀ᶠ n in l, ae_measurable (fs n) μ)
+  (h_bound : ∀ᶠ n in l, ∀ᵐ a ∂μ, ∥fs n a∥ ≤ bound a)
+  (bound_integrable : integrable bound μ)
+  (h_lim : ∀ᵐ a ∂μ, tendsto (λ n, fs n a) l (𝓝 (f a))) :
+  tendsto (λ n, set_to_fun hT (fs n)) l (𝓝 $ set_to_fun hT f) :=
+begin
+  rw tendsto_iff_seq_tendsto,
+  intros x xl,
+  have hxl, { rw tendsto_at_top' at xl, exact xl },
```

Comment:
I know you just copied this proof, but could you provide the type of `hxl` here, and the type of `h` on next line, to make this easier to read?